lib/rmg/CMapGenerator.cpp

```diff
@@ -491,14 +491,21 @@ const std::vector<HeroTypeID> CMapGenerator::getAllPossibleHeroes() const
 	for (HeroTypeID hero : map->getMap(this).allowedHeroes)
 	{
 		auto * h = dynamic_cast<const CHero*>(VLC->heroTypes()->getById(hero));
-		if ((h->onlyOnWaterMap && !isWaterMap) || (h->onlyOnMapWithoutWater && isWaterMap))
```

Comment:
This condition needs to stay. There are some heroes that are allowed **only** on water maps or maps without water (Ranloo, Kinkeria in HoTA).

Reply:
Not removed, but split into two. See 2 lines below.